source/core/constraint/unique.d

```diff
@@ -13,45 +13,177 @@ import core.sudoku.cell;
  */
 public class UniqueConstraint : Constraint
 {
-	private this(Cell[] cells...)
+	private this(Cell cell)
 	{
 		super (ConstraintType.Unique);
-		this.cells ~= cells;
+		this.cells ~= cell;
 	}
 
 
-	public static auto createSingle(Cell cell1, Cell cell2)
+	/** Connects two Cells
+	 *
+	 * Ensures `cell1` is connected to `cell2` and vice-versa
+	 * This means when a digit is checked for it's availability in `cell1`
+	 *     the validation is dependent of the digit in `cell2` and vice-versa
+	 * A Cell cannot connect to itself
+	 *
+	 * Params:
+	 *     cell1 = cell to connect
+	 *     cell2 = cell to connect
+	 *
+	 * Examples:
+	 * --------------------
+	 * Cell cell1 = new Cell(0);
+	 * Cell cell2 = new Cell(0);
+	 * UniqueConstraint.createSingle(cell1,cell2);
+	 * assert(cell1.get!UniqueConstraint.cells.equal([cell2]));
+	 * assert(cell2.get!UniqueConstraint.cells.equal([cell1]));
+	 * --------------------
+	 */
+	public static void createSingle(Cell cell1, Cell cell2)
 	{
-		if (cell1 is null || cell2 is null) return;
-		UniqueConstraint uc1 = new UniqueConstraint(cell1, cell2);
+		if (cell1 is null || cell2 is null)
+			return;
+
+		UniqueConstraint uc1 = new UniqueConstraint(cell2);
 		uc1.connect(cell1);
-		UniqueConstraint uc2 = new UniqueConstraint(cell1, cell2);
+		UniqueConstraint uc2 = new UniqueConstraint(cell1);
 		uc2.connect(cell2);
 	}
 
 
-	public static auto createMultiSingle(Cell cell, Cell[] cells...)
+	/** Connects a Cell to a group of Cells
+	 *
+	 * Ensures `cell` is connected to `cells` and that each Cell in `cells` is
+	 *     connected to `cell`
+	 * A Cell cannot connect to itself
+	 *
+	 * Params:
+	 *     cell = cell to connect
+	 *     cells = cells to be connected
+	 *
+	 * SeeAlso: void createSingle(**cell1**,**cell2**)
+	 */
+	public static void createMultiSingle(Cell cell, Cell[] cells...)
 	{
-		if (cell is null) return;
+		if (cell is null)
+			return;
+
 		foreach (Cell c; cells)
 		{
-			if (c is null) return;
+			if (c is null)
+				continue;
+
 			if (c != cell)
 				createSingle(cell, c);
 		}
 	}
 
 
-	public static auto createInterconnected(Cell[] cells...)
+	/** Connects group of Cells to each other
+	 *
+	 * Ensures all Cells in 'cells' are connected to each other
+	 * A Cell cannot connect to itself
+	 *
+	 * Params:
+	 *     cells = cells to connect to each other
+	 *
+	 * SeeAlso: void createSingle(**cell1**,**cell2**)
+	 */
+	public static void createInterconnected(Cell[] cells...)
 	{
+		import std.array : array;
+		import std.algorithm : filter;
+		if (canFind(cells, null))
+		{
+			cells = cells.filter!(cell => cell !is null).array;
```

Comment:
This is modifying the parameter, just wanna be sure this is intended behavior *(and if it is, please state that in the documentation)*

Reply:
Yes it is, if there are null objects in the variadic parameter, it removes them leaving only valid objects. I'll specify it in the documentation.